### ISSUE TITLE
App Rejected due to use of __NSCFBoolean non-public API

### ISSUE THREAD

**Issue by Shyam-Pgmr** [RestKit/RestKit#2508]
Hi,
  Apple rejected my app today due to use of __NSCFBoolean non-public API. They specified that  is being used within the app. 

**Full message from Apple**
Your app uses or references the following non-public APIs:

__NSCFBoolean

The use of non-public APIs is not permitted on the App Store because it can lead to a poor user experience should these APIs change.

Next Steps

To resolve this issue, please revise your app to remove any non-public APIs. If you have defined methods in your source code with the same names as the above-mentioned APIs, we suggest altering your method names so that they no longer collide with Apple's private APIs to avoid your app being flagged in future submissions.

Additionally, if you are using third-party libraries, please update to the most recent version of those libraries. If you do not have access to the libraries' source, you may be able to search the compiled binary using the "strings" or "otool" command line tools. The "strings" tool can output a list of the methods that the library calls and "otool -ov" will output the Objective-C class structures and their defined methods. These tools can help you narrow down where the problematic code resides. You could also use the "nm" tool to verify if any third-party libraries are calling these APIs.

**Comment by honkmaster** [comment]
@segiddins @valeriomazzeo Will this issue be addressed? From my short investigation `__NSCFBoolean` is always used in combination with `NSCFBoolean` e.g. in if-else such as: 

```
if ([propertyClass isSubclassOfClass:NSClassFromString(@"__NSCFBoolean")] || [propertyClass isSubclassOfClass:NSClassFromString(@"NSCFBoolean")]) {
    transformedValue = @([value boolValue]);
}
```

Would it be possible just to omit the first `isSubclassOfClass`?

**Comment by valeriomazzeo** [comment]
@honkmaster is this covered in the unit tests? If so, did you try running the tests after removing it the first condition?

**Comment by honkmaster** [comment]
I don't think it is fully covered, only for `RKObjectParameterization` an unit test exist. This one passes after removing the condition.

**Comment by mkujalowicz** [comment]
Perhaps one of the solutions could be replacing NSClassFromString(@"__NSCFBoolean") with [@YES class]

**Comment by honkmaster** [comment]
@valeriomazzeo Good to see the component is still alive due to recent commits. Do you already had time to look into this issue? I think it is quite important as any apps using Rest.Kit will not pass App Review.

**Comment by valeriomazzeo** [comment]
@honkmaster it's on my list, but haven't had time yet to dive into it. 
I have been investing it in updating the project to Xcode 8, instead.
